### ISSUE TITLE
Add internal Realm browser

### ISSUE THREAD

**Issue by kboyarshinov** [realm/realm-java#959]
Cocoa users have this feature. Would be good to have it for Android too. Right now we have to pull Realm database from device or emulator at first and then to view it in Cocoa browser (if you're on Mac).

I can suggest [AndroidDbInspector](https://github.com/infinum/android_dbinspector) library for SQL databases as a reference point. Realm browser can be implemented with browser Activity like AndroidDbInspector does. Better as a separate artifact from main realm-java.

**Comment by njovy** [comment]
Reading realm files on android devices sounds great. I thought that it would be great to have a realm browser feature implemented in [Stetho by facebook](https://github.com/facebook/stetho) as well. It's really easy to use and much better than using your phone to inspect sqlite files when you develop apps on your machine.

**Comment by jenzz** [comment]
stetho integration would be sweet :smile:

**Comment by cmelchior** [comment]
We definitely want to provide a Realm Browser for Android as well, however the exact details are still to be determined. As you point out, the Cocoa browser currently works for Android on Mac, albeit being a bit cumbersome.

Stetho looks very interesting, but as Realm isn't a SQL database, integration might not be straightforward. Also their license has some caveats that might make it problematic to use.

**Comment by zaki50** [comment]
I wrote [Realm module for Stetho](https://github.com/uPhyca/stetho-realm) a few days ago.

Currently documentation is only in Japanese.

**Comment by njovy** [comment]
@zaki50  Thanks for your hard work!!

**Comment by dmytrodanylyk** [comment]
> I can suggest AndroidDbInspector library for SQL databases as a reference point. Realm browser can be implemented with browser Activity like AndroidDbInspector does.

Check out my [Android Realm Database Browser](https://github.com/dmytrodanylyk/realm-browser). I am still working on [new features](https://github.com/dmytrodanylyk/realm-browser/issues).

**Comment by andrewthecoder** [comment]
Not sure if this is the best place to comment since the definition of "internal" Realm Browser is questionable, but I would love something cross-platform to at the very least view the contents of .realm files. If an IntelliJ IDEA  / Android Studio plugin were developed to allow reading Realm files as a "database" that would be even better, and cross-platform too!

**Comment by cmelchior** [comment]
@andrewthecoder Yes, right now it looks like that is the route we will be taking. As you point out it will be cross platform and with Google deprecating Eclipse, AndroidStudio is now the only official IDE. Tight IDE integration also has a lot of other benefits like making it possible to add things like custom lint checks and quick fixes.

That said, we haven't decided anything final yet, so things might still change.
